[SDS-468] Restore feature specs from Dashboards module

myabc opened the pull request:
# Ticket

https://community.openproject.org/wp/SDS-468

# What are you trying to accomplish?
<!-- Provide a description of the changes. -->

## Screenshots
<!-- Provide before/after screenshots, videos, or graphs for any visual changes; otherwise, remove this section -->

# What approach did you choose and why?
<!-- This section is a place for you to describe your thought process in making these changes.
     List any tradeoffs you made to take on or pay down tech debt.
     Describe any alternative approaches you considered and why you discarded them. -->

# Merge checklist

- [X] Added/updated tests
- [ ] Added/updated documentation in Lookbook (patterns, previews, etc)
- [ ] Tested major browsers (Chrome, Firefox, Edge, ...)

myabc committed:
Restore feature specs from Dashboards module

Moves feature specs and support files into Overviews module.

Also updates variable and file names from "overview" to "dashboard".

This partially reverts commit c3bbe97f81fce759166052f4c66a575fad0e2dc7.

myabc committed:
Restore feature specs from Dashboards module

Moves feature specs and support files into Overviews module.

Also updates variable and file names from "overview" to "dashboard".

This partially reverts commit c3bbe97f81fce759166052f4c66a575fad0e2dc7.

myabc committed:
[WIP] Use simpified Dashboard factory for features

myabc committed:
[WIP] Use simpified Dashboard factory for features

myabc committed:
Run former Dashboards feature specs with Selenium

myabc committed:
Run former Dashboards feature specs with Selenium

oliverguenther reviewed (approved):
I added a migration to rename the role_permission in the database, which would cause issues otherwise

myabc committed:
Fix RuboCop offenses in restored feature specs

Autocorrects Capybara/RSpec/HaveContent offenses introduced by
cops added on dev since the specs were written.
